src/Uno.UI/UI/Xaml/FrameworkElement.cs

```diff
@@ -883,6 +884,15 @@ internal virtual void UpdateThemeBindings(ResourceUpdateReason updateReason)
 				{
 					ActualThemeChanged?.Invoke(this, null);
 				}
+
+				if (this is Control @this && @this.GetTemplateRoot() is { } templateRoot)
+				{
+					var groups = VisualStateManager.GetVisualStateGroups(templateRoot);
+					foreach (var group in groups)
+					{
+						NativeDispatcher.Main.Enqueue(() => group.ReevaluateAppliedPropertySetters(@this));
```

Comment:
Why do you defer this on next dispatcher loop?

Reply:
BTW `NativeDispatcher.Main` is probably not right, you should use the dispatcher of `this`.

Reply:
@dr1rrb Trying to be as close as possible to WinUI, they keep enqueuing animation operations, then in next `Tick`, they call `FlushDeferredAnimationOperationQueue`

Reply:
This code is completely removed now.